pkg/filter/dt/transaction.go

```diff
@@ -27,112 +27,126 @@ import (
 	"github.com/cectc/dbpack/pkg/dt"
 	"github.com/cectc/dbpack/pkg/dt/api"
 	"github.com/cectc/dbpack/pkg/log"
+	"github.com/cectc/dbpack/pkg/tracing"
 )
 
 // handleHttp1GlobalBegin return bool, represent whether continue
-func (f *_httpFilter) handleHttp1GlobalBegin(ctx *fasthttp.RequestCtx, transactionInfo *TransactionInfo) (bool, error) {
+func (f *_httpFilter) handleHttp1GlobalBegin(ctx context.Context, fastHttpCtx *fasthttp.RequestCtx, transactionInfo *TransactionInfo) (bool, error) {
 	// todo support transaction isolation level
+	newCtx, span := tracing.GetTraceSpan(ctx, "http_filter_handle_global_session")
+	defer span.End()
 	transactionManager := dt.GetDistributedTransactionManager()
-	xid, err := transactionManager.Begin(ctx, string(ctx.Request.RequestURI()), transactionInfo.Timeout)
+	xid, err := transactionManager.Begin(newCtx, string(fastHttpCtx.Request.RequestURI()), transactionInfo.Timeout)
 	if err != nil {
-		ctx.Response.Reset()
-		ctx.SetStatusCode(http.StatusInternalServerError)
-		ctx.SetBodyString(fmt.Sprintf(`{"success":false,"error":"failed to begin global transaction, %v"}`, err))
+		tracing.RecordErrorSpan(span, err)
+		fastHttpCtx.Response.Reset()
+		fastHttpCtx.SetStatusCode(http.StatusInternalServerError)
+		fastHttpCtx.SetBodyString(fmt.Sprintf(`{"success":false,"error":"failed to begin global transaction, %v"}`, err))
 
 		return false, errors.Errorf("failed to begin global transaction, transaction info: %v, err: %v",
 			transactionInfo, err)
 	}
-	ctx.SetUserValue(XID, xid)
-	ctx.Request.Header.Add(XID, xid)
+	fastHttpCtx.SetUserValue(XID, xid)
+	fastHttpCtx.Request.Header.Add(XID, xid)
 	return true, nil
 }
 
-func (f *_httpFilter) handleHttp1GlobalEnd(ctx *fasthttp.RequestCtx) error {
-	xidParam := ctx.UserValue(XID)
+func (f *_httpFilter) handleHttp1GlobalEnd(ctx context.Context, fastHttpCtx *fasthttp.RequestCtx) error {
+	newCtx, span := tracing.GetTraceSpan(ctx, "http_filter_handle_global_session_end")
+	defer span.End()
+	xidParam := fastHttpCtx.UserValue(XID)
 	xid := xidParam.(string)
 
-	if ctx.Response.StatusCode() == http.StatusOK {
-		err := f.globalCommit(ctx, xid)
+	if fastHttpCtx.Response.StatusCode() == http.StatusOK {
+		err := f.globalCommit(newCtx, xid)
 		if err != nil {
+			tracing.RecordErrorSpan(span, err)
 			return errors.WithStack(err)
 		}
 	} else {
-		err := f.globalRollback(ctx, xid)
+		err := f.globalRollback(newCtx, xid)
```

Comment:
inside func `globalRollback`, add `newCtx, span := tracing.GetTraceSpan(ctx, "global_transaction_Rollback")`

Reply:
recommend to add spanName to the constant file